src/_pytest/logging.py

```
@@ -673,6 +673,8 @@ def pytest_runtest_logreport(self) -> None:
 
     def _runtest_for(self, item: nodes.Item, when: str) -> Generator[None, None, None]:
         """Implements the internals of pytest_runtest_xxx() hook."""
+        if self.log_level is None:
```

[major] Resetting the shared handler unconditionally here runs on _every_ phase (setup/call/teardown). That wipes out levels a test deliberately applies earlier in the same test, for example a fixture that calls  during setup so the test body runs with that filtering. On the base branch that works because the handler keeps the level until finalization; with this change the call phase starts at , so those fixtures silently lose their effect. Please gate the reset so it only runs for the first phase of each test (e.g. ), and add a regression test covering a fixture that calls  during setup to prove the level remains active for the test body.}